docs/HowToDevelopLoadableApps.md

```diff
@@ -0,0 +1,227 @@
+# How To Support Loadable Applications on TizenRT
+
+## CONTENTS
+
+- [Overview of Loadable Apps](#overview-of-loadable-apps)
+- [Assumptions & Limitations](#assumptions-&-limitations)
+- [Enable the existing loadable apps](#enable-the-existing-loadable-apps)
+- [Add new loadable apps configuration](#Add-new-loadable-apps-configuration)
+
+## Overview of Loadable Apps
+- When protected build (CONFIG_BUILD_PROTECTED) is selected, then a separate binary is generated for each user space application apart from a separate kernel binary.
+- There is an MPU based memory isolation & protection in place for each application. These apps are called loadable as they can be
+reloaded into their memory space even after fault detection.
+- Out of all the apps that can be loaded into the device, one of the apps could be critical whereas few can be non-critical.
+- Critical implies that in case of any type of fault detection in the app, the kernel has to be reloaded again (system reboot).
+- Non-critical implies that in case of any type of fault detection in the app, the app can be reloaded again without having the system to be rebooted again.
```

Comment:
Let us not mention these points about critical and non-critical apps since this might not be generic to all scenarios. 
Instead let us mention as below:
- TizenRt provides 2 sample loadable apps called micom and wifi
- In this example, micom is a critical, high priority app and wifi is non-critical low priority app
- All non critical threads such as tash console will be part of wifi app